### PR TITLE
Allen/add tutorials to znkHeader + tutorials

### PR DESCRIPTION
**Pull request by Allensy**

### CODE REVIEW

File: demo/tests/index.html [hunk 1]
```diff
@@ -48,7 +48,7 @@
     <script src="Chart.js/Chart.js"></script>
     <script src="angular-chart.js/dist/angular-chart.js"></script>
     <script src="v-accordion/dist/v-accordion.js"></script>
-    <script src="satellizer/dist/satellizer.js"></script>
```
Comment (devzinkerz) [inline demo/tests/index.html]: check if you updated satelizer to the new version -> make bower install before dist !!!

File: demo/tutorials/index.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+<script src="infra/dist/main.js"></script>
+<script src="ng-tags-input/ng-tags-input.min.js"></script>
+<script src="Chart.js/Chart.js"></script>
+<script src="angular-chart.js/dist/angular-chart.js"></script>
+<script src="v-accordion/dist/v-accordion.js"></script>
+<script src="satellizer/satellizer.js"></script>
+<script src="angular-drag-and-drop-lists/angular-drag-and-drop-lists.js"></script>
+<script src="angular-ui-grid/ui-grid.js"></script>
+<script src="elasticsearch/elasticsearch.js"></script>
+<script src="elasticsearch/elasticsearch.angular.js"></script>
```
Comment (devzinkerz) [inline demo/tutorials/index.html]: not needed

File: src/components/znkHeader/services/znkHeader.service.js [hunk 3]
```diff
@@ -37,7 +37,7 @@
 
                 addDefaultNavItem('ZNK_HEADER.WORKOUTS', 'app.workouts.roadmap', { reload: true });
                 addDefaultNavItem('ZNK_HEADER.TESTS', 'app.tests.roadmap');
-                // addDefaultNavItem('ZNK_HEADER.TUTORIALS', 'app.tutorials.roadmap');
+                addDefaultNavItem('ZNK_HEADER.TUTORIALS', 'app.tutorials');
```
Comment (devzinkerz) [inline src/components/znkHeader/services/znkHeader.service.js]: are you sure ?

File: src/components/tutorials/services/tutorials.service.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                    });
+                };
+
+                TutorialsSrv.getSubjectOrder = function () {
+                    if (!_subjectOrderGetter) {
+                        var errMsg = 'Tutorials Service: subjectOrderGetter was not set.';
+                        $log.error(errMsg);
+                        return $q.reject(errMsg);
+                    }
+                    return $q.when($injector.invoke(_subjectOrderGetter));
```
Comment (devzinkerz) [inline src/components/tutorials/services/tutorials.service.js]: add check angular.isFunction before invoke

File: src/components/tutorials/services/tutorials.service.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                    });
+                };
+
+                TutorialsSrv.getTutorial = function (tutorialId) {
+                    return StorageRevSrv.getContent({
+                        exerciseId: tutorialId, exerciseType: 'tutorial'
+                    });
+                };
+
+                TutorialsSrv.getAllTutorials = function() {
```
Comment (devzinkerz) [inline src/components/tutorials/services/tutorials.service.js]: Does someone uses  it ? all tutorial?

File: src/components/tutorials/services/tutorials.service.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
+(function (angular) {
+    'use strict';
+    angular.module('znk.infra-web-app.tutorials').provider('TutorialsSrv', [
```
Comment (devzinkerz) [inline src/components/tutorials/services/tutorials.service.js]: add test

File: src/components/tutorials/services/tutorials.service.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                            });
+                        });
+
+                        return $q.all(allProm).then(function () {
+                            return tutorialHeaders;
+                        });
+                    });
+                };
+
+                TutorialsSrv.getSubjectOrder = function () {
```
Comment (devzinkerz) [inline src/components/tutorials/services/tutorials.service.js]: maybe it should be global

File: src/components/tutorials/module.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        'ngMaterial',
+        'pascalprecht.translate',
+        'znk.infra.enum',
+        'znk.infra.config',
+        'znk.infra.general',
+        'znk.infra.svgIcon',
+        'ui.router',
+        'znk.infra-web-app.diagnostic',
+        'znk.infra-web-app.completeExercise',
+        'znk.infra.scoring',
```
Comment (devzinkerz) [inline src/components/tutorials/module.js]: are you sure scoring is needed ? in infra?

File: src/components/tutorials/controllers/tutorialsRoadmap.controller.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra-web-app.tutorials').controller('TutorialsRoadmapController',
+        function ($translatePartialLoader, tutorials) {
+            'ngInject';
+            $translatePartialLoader.addPart('tutorials');
+            var vm = this;
+            vm.tutorials = tutorials;
+            vm.activeSubjectChanged = function () {
```
Comment (devzinkerz) [inline src/components/tutorials/controllers/tutorialsRoadmap.controller.js]: why do you need empty func?

File: src/components/tutorials/components/tutorialPane/tutorialPane.component.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        var subjectOrderProm = TutorialsSrv.getSubjectOrder();
+        vm.subjectsMap = SubjectEnum.getEnumMap();
+
+        vm.$onInit = function () {
+            $q.all([
+                subjectOrderProm
+            ]).then(function (res) {
+                vm.subjecstOrder = res[0];
+                if (!vm.activeSubject) {
+                    vm.activeSubject = res[0][0];
```
Comment (devzinkerz) [inline src/components/tutorials/components/tutorialPane/tutorialPane.component.js]: why not subjecstOrder[0] ?

File: src/components/tutorials/components/tutorialPane/tutorialPane.component.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+                    vm.ngModelCtrl.$setViewValue(+res[0][0]);
```
Comment (devzinkerz) [inline src/components/tutorials/components/tutorialPane/tutorialPane.component.js]: subjecstOrder[0]

Comment (devzinkerz) [inline src/components/tutorials/components/tutorialPane/tutorialPane.component.js]: why in [0] location ?

File: src/components/znkHeader/services/znkHeader.service.js (hunk 3 above)
Comment (Allensy) [inline src/components/znkHeader/services/znkHeader.service.js]: I am sure

File: src/components/tutorials/services/tutorials.service.js (hunk 5 above)
Comment (Allensy) [inline src/components/tutorials/services/tutorials.service.js]: yes

File: src/components/tutorials/services/tutorials.service.js (hunk 7 above)
Comment (Allensy) [inline src/components/tutorials/services/tutorials.service.js]: it is now

File: src/components/tutorials/services/tutorials.service.js (hunk 4 above)
Comment (Allensy) [inline src/components/tutorials/services/tutorials.service.js]: Done

File: src/components/tutorials/module.js (hunk 8 above)
Comment (Allensy) [inline src/components/tutorials/module.js]: removed

File: src/components/tutorials/controllers/tutorialsRoadmap.controller.js (hunk 9 above)
Comment (Allensy) [inline src/components/tutorials/controllers/tutorialsRoadmap.controller.js]: removed

File: src/components/tutorials/components/tutorialPane/tutorialPane.component.js (hunk 10 above)
Comment (Allensy) [inline src/components/tutorials/components/tutorialPane/tutorialPane.component.js]: You're right! fixed.

File: src/components/tutorials/components/tutorialPane/tutorialPane.component.js (hunk 11 above)
Comment (Allensy) [inline src/components/tutorials/components/tutorialPane/tutorialPane.component.js]: first subject by default

Comment (Allensy) [inline src/components/tutorials/components/tutorialPane/tutorialPane.component.js]: fixed

**Pull request merged**